pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1126,8 +1126,25 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 	if len(b.iface.Ports) == 0 {
 		err = Handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
 			"counter", "dnat", "to", b.getVifIpByProtocol(proto))
+		if err != nil {
+			return err
+		}
 
-		return err
+		err = Handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND",
```

Comment:
in reaction to [Alona's comment](https://github.com/kubevirt/kubevirt/pull/5200#discussion_r606840280)
I updated this rule (same rule, only for VM with explicit ports) in my PR like this

```
err = Handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND",
	strings.ToLower(port.Protocol),
	"dport",
	strconv.Itoa(int(port.Port)),
	Handler.GetNFTIPString(proto), "daddr", b.getVifIpByProtocol(proto),
	"counter", "snat", "to", b.getGatewayByProtocol(proto))
```

Which matches packets by destination IP (being equal to VM IP) and it eliminates danger of matching packets with external IP. After all, we want to `snat` only packets that are going into the VM, and at `postrouting` stage, these packets have the destionation IP address set to VM IP, because of the `dnat` rule in the `output` chain which happens earlier. 

cc @AlonaKaplan

Reply:
understand. Makes more sense. I changed the rule to what you suggested without the `dport` part. because the point is to not be depended on specific port.

Reply:
> I changed the rule to what you suggested without the dport part.

and I totally missed that!